asset-transfer-ledger-queries/application-javascript/app.js

```diff
@@ -0,0 +1,213 @@
+}
+
+// pre-requisites:
+// - fabric-sample two organization test-network setup with two peers, ordering service,
+//   and 2 certificate authorities, with the state database using couchdb
+//         ===> from directory /fabric-samples/test-network
+//         ./network.sh up createChannel -ca -s couchdb
+// - Use any of the asset-transfer-basic chaincodes deployed on the channel "mychannel"
+//   with the chaincode name of "ledger". The following deploy command will package,
+//   install, approve, and commit the javascript chaincode, all the actions it takes
+//   to deploy a chaincode to a channel.
+//         ===> from directory /fabric-samples/test-network
+//         ./network.sh deployCC -ccn ledger -ccl javascript
+// - Be sure that node.js is installed
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-javascript
+//         node -v
+// - npm installed code dependencies
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-javascript
+//         npm install
+// - to run this test application
+//         ===> from directory /fabric-samples/asset-transfer-basic/application-javascript
+//         node app.js
+
+// NOTE: If you see this error:
+//
+//  2020-08-07T20:23:17.590Z - error: [DiscoveryService]: send[mychannel] - Channel:mychannel received discovery error:access denied
+//  ******** FAILED to run the application: Error: DiscoveryService: mychannel error: access denied
+//
+// Delete the /fabric-samples/asset-transfer-basic/application-javascript/wallet directory
+// and retry this application.
+//
+// The certificate authority must have been restarted and the saved certificates for the
+// admin and application user are not valid. Deleting the wallet store will force these to be reset
+// with the new certificate authority.
+//
+
+/**
+ *  A test application to show basic operations with any of the asset-transfer-basic chaincodes
+ *   -- How to submit a transaction
+ *   -- How to query and check the results
+ *
+ * To see the SDK workings, try setting the logging to show on the console before running
+ *        export HFC_LOGGING='{"debug":"console"}'
+ */
+async function main() {
+	try {
+		// build an in memory object with the network configuration (also known as a connection profile)
+		const ccp = buildCCP();
+
+		// build an instance of the fabric ca services client based on
+		// the information in the network configuration
+		const caClient = buildCAClient(FabricCAServices, ccp);
+
+		// setup the wallet to hold the credentials of the application user
+		const wallet = await buildWallet(Wallets, walletPath);
+
+		// in a real application this would be done on an administrative flow, and only once
+		await enrollAdmin(caClient, wallet);
+
+		// in a real application this would be done only when a new user was required to be added
+		// and would be part of an administrative flow
+		await registerUser(caClient, wallet, userId, 'org1.department1');
```

Comment:
I realize it's part of the common code, but would be more clear if this was named 'registerAndEnrollUser(). Could be done in follow-on PR.

Reply:
done